### ISSUE TITLE
[data grid]: Keyboard navigation does not support row dragging.

### ISSUE THREAD

**Issue by naidusaiman** [mui/mui-x#20586]
### Steps to reproduce

Steps:
1. https://mui.com/x/react-data-grid/row-ordering/
2. Try Row Drag and Drop using keyboard

### Current behavior

Current: Unable to drag and drop rows using keyboard navigation

### Expected behavior

Expected: Should be able to drag and drop rows using keyboard navigation

### Your environment

<details>
  <summary><code>npx @mui/envinfo</code></summary>

```
  Don't forget to mention which browser you used.
  Output from `npx @mui/envinfo` goes here.
```
</details>

**Search keywords**: DataGrid RowDrag Accessibility

**Comment by michelengelen** [comment]
Hey @naidusaiman and thanks for opening this issue.

Your request seems valid, but I wonder how you would expect this to work. There is no clear definition on how dragging should work. Could you clarify this for us?

**Comment by naidusaiman** [comment]
Hello @michelengelen 
Ideally, it would be beneficial to support behavior similar to what libraries such as `react-dnd` provide—typically using a modifier key (Space) in combination with the Up/Down arrow keys to move items.

I understand that, in the context of the DataGrid, the Space key is currently bound to navigating to the last row. In that case, we could consider using an alternative key to achieve similar interaction semantics.

**Comment by MBilalShafi** [comment]
@naidusaiman Thank you for the input. I've marked the issue as a `waiting for 👍` to gauge the community demand for this particular feature vs others in pipeline.

You can add a 👍 emojie reaction to the issue description to emphasize the demand. If more such upvotes are added, the feature will be more likely to be picked up by someone from the team.

To speed up even further, you can shoot up a PR yourself. If you need any help doing so, you can reach out to me or @michelengelen and we'll be glad to help.
